Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp

```diff
@@ -429,12 +445,36 @@ namespace ROS2
         }
     }
 
+    AZStd::string JointsManipulationComponent::GetManipulatorNamespace()
+    {
+        auto* frameComponent = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(m_entity);
+        AZ_Assert(frameComponent, "ROS2FrameComponent is required for joints.");
+        if (frameComponent)
+        {
+            return frameComponent->GetNamespace();
+        }
+        return "";
+    }
+
     void JointsManipulationComponent::OnTick(float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
     {
         if (m_manipulationJoints.empty())
         {
+            const AZStd::string manipulatorNamespace = GetManipulatorNamespace();
+            AZ_Printf("JointsManipulationComponent", "Initializing joints manipulation component for namespace %s", manipulatorNamespace.c_str());
+            AZStd::unordered_map<AZStd::string, JointPosition> intialPositonNamespaced;
+            AZStd::transform(
+                m_initialPositions.begin(),
+                m_initialPositions.end(),
+                AZStd::inserter(intialPositonNamespaced, intialPositonNamespaced.end()),
+                [&manipulatorNamespace](const auto& pair)
+                {
+                    return AZStd::make_pair(ROS2::ROS2Names::GetNamespacedName(manipulatorNamespace, pair.first), pair.second);
+                });
+
             m_manipulationJoints = Internal::GetAllEntityHierarchyJoints(GetEntityId());
-            Internal::SetInitialPositions(m_manipulationJoints, m_initialPositions);
+
+            Internal::SetInitialPositions(m_manipulationJoints, intialPositonNamespaced);
```

Comment:
Please add documentation to this structure (m_manipulationJoints) that it includes namespaces.

Raise an issue to remove namespaces from everything except interfacing with ROS.

Reply:
Added doc to private member. Raised an issue https://github.com/o3de/o3de-extras/issues/502